MODULE.bazel

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Bazel module configuration for IREE."""
+
+module(
+    name = "iree_core",
+    version = "0.0.1",
+)
+
+# Core Bazel dependencies - versions aligned with LLVM's MODULE.bazel
+bazel_dep(name = "apple_support", version = "1.24.1", repo_name = "build_bazel_apple_support")
+bazel_dep(name = "bazel_skylib", version = "1.8.2")
+bazel_dep(name = "platforms", version = "1.0.0")
+bazel_dep(name = "rules_cc", version = "0.2.11")
+bazel_dep(name = "rules_python", version = "1.6.3")
+bazel_dep(name = "rules_shell", version = "0.6.1")
+
+# Required by LLVM (for clang-fuzzer tools)
+bazel_dep(name = "protobuf", version = "31.1", repo_name = "com_google_protobuf")
```

Comment:
I don't think protobuf should be needed for IREE's LLVM build. I left a comment at https://github.com/llvm/llvm-project/pull/168928#discussion_r2566681186.

(If it is relatively painless to add here, fine, but I still think we should push back)

Reply:
I can try removing it. Bzlmod should only download it anyway if there is a real dependency on it.